### ISSUE TITLE
gd_gd2.c build fails: error: 'INT_MAX' undeclared

### ISSUE THREAD

**Issue by totemofwolf** [libgd/libgd#387]
Compile errors On Centos 6 box :
```
gd_gd2.c: In function '_gd2GetHeader':
gd_gd2.c:212:40: error: 'INT_MAX' undeclared (first use in this function)
   if (*ncx <= 0 || *ncy <= 0 || *ncx > INT_MAX / *ncy) {
                                        ^
gd_gd2.c:212:40: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [gd_gd2.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory `/data/home/rpmbuild/rpmbuild/BUILD/libgd-2.2.4/src'
make[1]: *** [all] Error 2
make[1]: Leaving directory `/data/home/rpmbuild/rpmbuild/BUILD/libgd-2.2.4/src'
make: *** [all-recursive] Error 1
error: Bad exit status from /home/rpmbuild/rpmbuild/tmp/rpm-tmp.drxAFz (%prep)
```

**Comment by vapier** [comment]
this build failure doesn't make sense.  `INT_MAX` is defined by `limits.h` and `gd_gd2.c` includes that unconditionally at the top of the file.

please check your system to make sure it's not broken.  you can try something like:
`gcc -E -dD - <<<'#include <limits.h>' | grep -e INT_MAX -e limits.h`

**Comment by totemofwolf** [comment]
after insert '#include <limits.h>' to  gd_gd2.c，build done.

`  
  75 /* 2.0.29: no more errno.h, makes windows happy */
  76 #include <math.h>
  77 #include <string.h>
  78 #include "gd.h"
  79 #include "gd_errors.h"
  80 #include "gdhelpers.h"
  81 
+  82 #include <limits.h>
  83 
`

**Comment by vapier** [comment]
i was looking at the branch rather than the 2.2.4 release.  this is already fixed there by commit 381e89de16125bb0e86ffc144d6fe3e9cfdd898f.

**Comment by VincentN88888** [comment]
I tried your command `gcc -E -dD - <<<'#include <limits.h>' | grep -e INT_MAX -e limits.h`
On mac server 10.5 
but I got that. not knowing if it's broken or not, or how I could fixed the build error
#define __INT_MAX__ 2147483647
# 1 "/usr/lib/gcc/i686-apple-darwin9/4.0.1/include/limits.h" 1 3 4
# 1 "/usr/lib/gcc/i686-apple-darwin9/4.0.1/include/syslimits.h" 1 3 4
# 8 "/usr/lib/gcc/i686-apple-darwin9/4.0.1/include/limits.h" 2 3 4
# 1 "/usr/include/limits.h" 1 3 4
# 61 "/usr/include/limits.h" 3 4
# 64 "/usr/include/limits.h" 2 3 4
# 1 "/usr/include/machine/limits.h" 1 3 4
# 1 "/usr/include/i386/limits.h" 1 3 4
# 37 "/usr/include/i386/limits.h" 3 4
# 1 "/usr/include/i386/_limits.h" 1 3 4
# 23 "/usr/include/i386/_limits.h" 3 4
# 41 "/usr/include/i386/limits.h" 2 3 4
# 59 "/usr/include/i386/limits.h" 3 4
#define UINT_MAX 0xffffffff
#define INT_MAX 2147483647
# 9 "/usr/include/machine/limits.h" 2 3 4
# 65 "/usr/include/limits.h" 2 3 4
# 1 "/usr/include/sys/syslimits.h" 1 3 4
# 66 "/usr/include/sys/syslimits.h" 3 4
# 75 "/usr/include/sys/syslimits.h" 3 4
# 66 "/usr/include/limits.h" 2 3 4
# 11 "/usr/lib/gcc/i686-apple-darwin9/4.0.1/include/limits.h" 2 3 4
# 51 "/usr/lib/gcc/i686-apple-darwin9/4.0.1/include/limits.h" 3 4
#define INT_MIN (-INT_MAX - 1)
#undef INT_MAX
#define INT_MAX __INT_MAX__
#undef UINT_MAX
#define UINT_MAX (INT_MAX * 2U + 1U)
# 106 "/usr/lib/gcc/i686-apple-darwin9/4.0.1/include/limits.h" 3 4
